### PR TITLE
person-task

### PR DESCRIPTION
**Pull request by AkhtharBawazeer**
2 Classes person and client
[ADD] inheritance module

### CODE REVIEW

File: person/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    'name': 'Person and Client Management',
+    'version': '1.0',
```
Comment (MORPSmcaf) [inline person/__manifest__.py]: rename to '16.0.1.0.0',

File: person/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    'summary': 'Manage information about persons and clients',
+    'author': 'Akhthar',
+    'website': 'https://www.Test.com',
```
Comment (MORPSmcaf) [inline person/__manifest__.py]: test to lover case

File: person/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+    'name': 'Person and Client Management',
+    'version': '1.0',
+    'summary': 'Manage information about persons and clients',
+    'author': 'Akhthar',
```
Comment (MORPSmcaf) [inline person/__manifest__.py]: 'MCAF'

File: person/controllers/__init__.py
Comment (MORPSmcaf) [inline person/controllers/__init__.py:1]: remove controllers folder with file you will not use it

File: person/demo/demo.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<odoo>
+    <data>
+<!--
```
Comment (MORPSmcaf) [inline person/demo/demo.xml]: remove comments

File: person/models/client.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from odoo import models, fields
+
+class Person_inher(models.Model):
+    #_name = 'person.inherit'
+    _description = 'Model of a client'
+    _inherit = 'person.person'
+
+    # Additional fields specific to clients
```
Comment (MORPSmcaf) [inline person/models/client.py]: remove comments

File: person/models/client.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+from odoo import models, fields
+
+class Person_inher(models.Model):
```
Comment (MORPSmcaf) [inline person/models/client.py]: rename class as a CustomClient

File: person/models/person.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
+import re
+
+
+class Person(models.Model):
```
Comment (MORPSmcaf) [inline person/models/person.py]: (models.AbstractModel)

File: person/models/person.py [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    _name = 'person.person'
```
Comment (MORPSmcaf) [inline person/models/person.py]: abstract.person

File: person/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    'website': 'https://www.Test.com',
+    'license': 'LGPL-3',
+    'depends': ['base'],
+    'application': True,
+    'installable': True,
+    'data': [
+        'security/ir.model.access.csv',
+        'views/person.xml',
```
Comment (MORPSmcaf) [inline person/__manifest__.py]: rename abstract_person.xml

File: person/models/__init__.py [hunk 10]
```diff
@@ -0,0 +1,2 @@
+from . import person
```
Comment (MORPSmcaf) [inline person/models/__init__.py]: rename to abstract_person

File: person/models/client.py [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+    #_name = 'person.inherit'
```
Comment (MORPSmcaf) [inline person/models/client.py]: _name = 'custom.client'

File: person/models/client.py [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+    _description = 'Model of a client'
+    _inherit = 'person.person'
```
Comment (MORPSmcaf) [inline person/models/client.py]: abstract.person

File: person/models/client.py (hunk 11 above)
Comment (AkhtharBawazeer) [reply person/models/client.py]: Class name to AbstractPerson

File: person/models/__init__.py (hunk 10 above)
Comment (AkhtharBawazeer) [reply person/models/__init__.py]: done

File: person/models/client.py (hunk 11 above)
Comment (AkhtharBawazeer) [reply person/models/client.py]: done

File: person/models/client.py (hunk 12 above)
Comment (AkhtharBawazeer) [reply person/models/client.py]: done

File: person/__manifest__.py [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+    'summary': 'Manage information about persons and clients',
+    'author': 'MCAF',
+    'website': 'https://www.test.com',
+    'license': 'LGPL-3',
+    'depends': ['base'],
+    'application': True,
+    'installable': True,
+    'data': [
+        'security/ir.model.access.csv',
+        'views/abstract_person.xml',
```
Comment (MORPSmcaf) [inline person/__manifest__.py]: views/custom_client_view.xml'

File: person/views/abstract_person.xml [hunk 14]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_person_form" model="ir.ui.view">
```
Comment (MORPSmcaf) [inline person/views/abstract_person.xml]: custom_client_form_view"

File: person/views/abstract_person.xml
Comment (MORPSmcaf) [inline person/views/abstract_person.xml:1]: custom_client_view.xml

File: person/views/abstract_person.xml [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+        <field name="name">person.person.form</field>
```
Comment (MORPSmcaf) [inline person/views/abstract_person.xml]: custom.client.form<

File: person/views/abstract_person.xml [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+        <field name="model">custom.client</field>
+        <field name="arch" type="xml">
+            <form string="Person">
```
Comment (MORPSmcaf) [inline person/views/abstract_person.xml]: string="Client"

File: person/views/abstract_person.xml [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+                        <field name="gender" widget="selection" required="1"/>
+
+                    </group>
+                </sheet>
+            </form>
+        </field>
+    </record>
+
+
+    <record id="view_person_tree" model="ir.ui.view">
```
Comment (MORPSmcaf) [inline person/views/abstract_person.xml]: custom_client_tree_view

File: person/views/abstract_person.xml [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+        <field name="name">person.person.tree</field>
+        <field name="model">custom.client</field>
+        <field name="arch" type="xml">
+            <tree string="Person">
```
Comment (MORPSmcaf) [inline person/views/abstract_person.xml]: string="Client"

File: person/views/abstract_person.xml [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+                <field name="email" required="1"/>
+                <field name="photo" required="1"/>
+                <field name="gender" required="1"/>
+            </tree>
+        </field>
+    </record>
+
+
+    <record id="action_client_example" model="ir.actions.act_window">
+        <field name="name">client name</field>
```
Comment (MORPSmcaf) [inline person/views/abstract_person.xml]: Client name

File: person/views/abstract_person.xml [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+        <field name="res_model">custom.client</field>
+        <field name="view_mode">tree,form</field>
+    </record>
+
+    <menuitem id="menu_root_person"
+              name="Person_client"/>
```
Comment (MORPSmcaf) [inline person/views/abstract_person.xml]: name="Custom Client"

File: person/views/abstract_person.xml (hunk 17 above)
Comment (AkhtharBawazeer) [reply person/views/abstract_person.xml]: done

File: person/views/abstract_person.xml (hunk 18 above)
Comment (AkhtharBawazeer) [reply person/views/abstract_person.xml]: done

File: person/views/abstract_person.xml (hunk 19 above)
Comment (AkhtharBawazeer) [reply person/views/abstract_person.xml]: done

File: person/views/abstract_person.xml (hunk 20 above)
Comment (AkhtharBawazeer) [reply person/views/abstract_person.xml]: Done

**Review by MORPSmcaf — APPROVED** [review]

**Pull request merged**